tt_metal/hw/inc/dataflow_api.h

```diff
@@ -1486,6 +1499,70 @@ void noc_async_write_multicast_loopback_src(
     WAYPOINT("NMLD");
 }
 
+/**
+ * Initiates an asynchronous write from a source address in L1 memory on the
+ * Tensix core executing this function call to an L- shaped destination which is defined by 
+ * a grid and an exclusion zone. 
+ * The destinations are specified using a uint64_t encoding referencing an
+ * on-chip grid of nodes located at NOC coordinate range
+ * (x_start,y_start,x_end,y_end) and a local address created using
+ * *get_noc_multicast_addr* function. Also, *see noc_async_write_barrier*.
+ * Similarly, the exclusion zone is specified using uint32_t encoding referencing
+ * an on-chip and directions relative to it created using *get_noc_exclude_addr* function.
+ *
+ * The destination nodes can only be a set of Tensix cores + L1 memory address.
+ * The destination nodes must form a rectangular grid. The destination L1
+ * memory address must be the same on all destination nodes.
+ *
+ * With this API, the multicast sender cannot be part of the multicast
+ * destinations. If the multicast sender has to be in the multicast
+ * destinations (i.e. must perform a local L1 write), the other API variant
+ * *noc_async_write_multicast_loopback_src* can be used.
+ *
+ * Note: The number of destinations needs to be non-zero. Besides that,
+ * there is no restriction on the number of destinations, i.e. the
+ * multicast destinations can span the full chip. However, as mentioned
+ * previously, the multicast source cannot be part of the destinations. So, the
+ * maximum number of destinations is 119.
```

Comment:
max being 119 is incorrect as well

Reply:
It isn't correct for the other multicast methods either. I'll rewrite